flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/restart/FailureRateRestartStrategy.java

```diff
@@ -35,19 +34,21 @@
  * with a fixed time delay in between.
  */
 public class FailureRateRestartStrategy implements RestartStrategy {
-	private final int maxFailuresPerUnit;
-	private final TimeUnit failureRateUnit;
-	private final long delayBetweenRestartAttempts;
-	private List<Long> restartTimestamps = new ArrayList<>();
+	private final Duration failuresInterval;
+	private final Duration delayInterval;
+	private EvictingQueue<Long> restartTimestampsQueue;
```

Comment:
Can we replace `EvictingQueue` with an `ArrayDequeue`? It gives you almost the same features as the `EvictingQueue` with similar performance and we would not introduce a further dependency on Guava. The reason is that we gradually try to get rid of Guava because different versions of Guava are not fully compatible.